### PR TITLE
Update tiflash-build-common.groovy

### PR DESCRIPTION
**Pull request by solotzg**
add cache for rustup toolchains

### CODE REVIEW

File: jenkins/pipelines/ci/tiflash/tiflash-build-common.groovy [hunk 1]
```diff
@@ -491,6 +491,14 @@ def prepareStage(repo_path) {
                     rm -rf ~/.cargo/registry/index && ln -s /home/jenkins/agent/rust/registry/index ~/.cargo/registry/index 
                     rm -rf ~/.cargo/git/db && ln -s /home/jenkins/agent/rust/git/db ~/.cargo/git/db
                     rm -rf ~/.cargo/git/checkouts && ln -s /home/jenkins/agent/rust/git/checkouts ~/.cargo/git/checkouts
+                    
+                    rm -rf ~/.rustup/tmp
+                    rm -rf ~/.rustup/toolchains
+                    mkdir -p /home/jenkins/agent/rust/rustup-env/tmp
+                    mkdir -p /home/jenkins/agent/rust/rustup-env/toolchains
+                    ln -s /home/jenkins/agent/rust/rustup-env/tmp /root/.rustup/tmp
+                    ln -s /home/jenkins/agent/rust/rustup-env/toolchains /root/.rustup/toolchains
```
Comment (solotzg) [inline jenkins/pipelines/ci/tiflash/tiflash-build-common.groovy:500]:
```suggestion
                    rm -rf ~/.rustup/tmp
                    rm -rf ~/.rustup/toolchains
                    mkdir -p /home/jenkins/agent/rust/rustup-env/tmp
                    mkdir -p /home/jenkins/agent/rust/rustup-env/toolchains
                    ln -s /home/jenkins/agent/rust/rustup-env/tmp ~/.rustup/tmp
                    ln -s /home/jenkins/agent/rust/rustup-env/toolchains ~/.rustup/toolchains
```

**Comment by solotzg** [comment]
@SchrodingerZhu @purelind PTAL

**Comment by solotzg** [comment]
Verified in https://ci.pingcap.net/blue/organizations/jenkins/tiflash-build-common/detail/tiflash-build-common/6232/pipeline

**Review by purelind — APPROVED** [review]
LGTM

**Pull request merged**